### ISSUE TITLE
Add support for helper platforms

### ISSUE THREAD

**Issue by postlund** [rospogrigio/localtuya#253]
We should add generic support for the helper platforms, e.g. everyone named `input_` to support the various odd-cases we might find.

**Comment by ultratoto14** [comment]
@postlund can you describe a bit more what you're thinking about, or just put an example ?

**Comment by postlund** [comment]
@ultratoto14 An imaginary example would be a DP that expose the current speed of a fan as an integer and allows changing the speed by setting a new value. By providing an implementation of [input_number](https://www.home-assistant.io/integrations/input_number/), we can support those kinds of use cases. I happen to own a LED string from DELTACO that supports eight different light modes (scenes) which are just represented by an integer. `input_number` would work for that.

Not all helpers are relevant as some are implemented best by another platform, e.g. `switch` can be used instead of `input_boolean`.

**Comment by ultratoto14** [comment]
Ok, i was thinking of the same but using theblueprint feature. 
Do you imagine internal mapping in that case ? 
Displayed value and sent value for each entry ?

**Comment by postlund** [comment]
I'm not sure how blueprint fits the picture here? You mean like use a sensor in conjunction with the `set_dp` service and hide it behind some config in the blueprint?

I just imagine that you specify min, max and step length more or less when adding the entity and let the DP value map to that range (with some error handling I case there's a mismatch). When changing the slider/using increase or decrease, that value would be written to the DP. Nothing fancy-schmancy.

**Comment by ultratoto14** [comment]
I was thinking of the input select with different scenes, it's like a combo, I would in that case, prefer having a combo with scene names than just numbers, that may send simple or complex data using set DP.

**Comment by ultratoto14** [comment]
Yes for sliders(input_number), totally following you

**Comment by postlund** [comment]
Yeah, `input_select` is a bit more complex as we are limited in what we can do. We have a few alternatives as I see it:

* Give in and allow some kind of string format, e.g. `Night=1,Day=2`
* Do some kind of simple "generic" editor based on the concept I used for "subscribed events" in the `remote_homeassistant` component. It's hard to explain, it's easier to try out for yourself if you want to see how it works (add a remote instance and go to the last page under Options).
* Add a websocket or REST API for managing scenes and create a custom Lovelace card for it. Most flexible but also feels most out of place.
* Allow imports of well-formed data from a URL (similarly what we have been talking about with the database)

**Comment by ultratoto14** [comment]
Aaahhh, didn't know you are also working on remote homeassistant, it seems that it's not the first time you saved my life 😄 

Will check tomorrow

**Comment by postlund** [comment]
But of course I do 😉

**Comment by ultratoto14** [comment]
I like the way `remote_homeassistant` let you add events. Great job again.
Maybe a mix of your two first points, the way it's done in `remote_homeassistant` and `Night=1` when we add an entry, it may be optional, we can use `1` or `Night=1`
